pkg/storage/redis/redis.go

```diff
@@ -66,13 +68,18 @@ func New(address, recordType string, deletePermanentAfter int64) (*DB, error) {
 		deletedSet:             recordType + "_deleted_set",
 		lastVersionKey:         recordType + "_last_version",
 	}
+	metrics.AddRedisMetrics(db.pool.Stats)
 	return db, nil
 }
 
 // Put sets new record for given id with input data.
-func (db *DB) Put(ctx context.Context, id string, data *anypb.Any) error {
+func (db *DB) Put(ctx context.Context, id string, data *anypb.Any) (err error) {
```

Comment:
Maybe this is more of a preference thing but I find named result parameters to hurt readability (though I understand why you have one here, in order to support the early defer statement), in all but very short functions. 

[Effective Go](https://golang.org/doc/effective_go.html#named-results) is the only thing I've found to weigh in on this though.

Reply:
Yes I _think_ we can do without it, but it makes the actual error variable easier to lose track of.  

I know there's a lot of animosity towards naked returns but this seems like a good use of named parameters.